### PR TITLE
Http server

### PR DESCRIPTION
**Pull request by saladuit**
- [ ] Added inclusion dependency checks
 - [ ] Integrated ConfigParser class into the HTTPServer class
 - [ ] Added logging to file
 - [ ] Made a ThreadPool class but decided to postpone the integration until we finish the project
 - [ ] Made a very simple test script (@Tentanus you will surely upgrade that!)
 - [ ] Made a main with a default usage of our server

### CODE REVIEW

**Review by Tentanus — COMMENTED** [review]
minor const/ref additions

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,9 @@
 !.gitignore
 compile_commands.json
 
+# Logger
+log.txt
```
[the remark this answers is not in the document]
Comment (saladuit) [reply .gitignore:7]: I only didn't create the log folder. It is being dumped into build.

File: include/ConfigParser.hpp [hunk 2]
```diff
@@ -67,7 +67,7 @@ class ConfigParser
 
 	void readConfig();
 
-	std::map<GlobalSetting, std::string> getGlobalSettings() const;
+	std::string getGlobalSettings(GlobalSetting setting) const;
```
[the remark this answers is not in the document]
Comment (saladuit) [reply include/ConfigParser.hpp:70]: Done

File: include/Logger.hpp [hunk 3]
```diff
@@ -33,18 +34,26 @@ class Logger
...
 	void log(LogLevel lvl, const std::string &message, Args... args)
 	{
+		std::string formatted_msg = format(message, args...);
 		if (lvl < _current_level)
 			return;
-
-		std::string formatted_msg = format(message, args...);
+		if (!_log_file.is_open())
+		{
+			std::cout << "Failed to open log file" << std::endl;
```
[the remark this answers is not in the document]
Comment (saladuit) [reply include/Logger.hpp:42]: done

File: include/Logger.hpp [hunk 4]
```diff
@@ -33,18 +34,26 @@ class Logger
...
 		std::cout << "[" << getTimestamp() << "] " << logLevelToString(lvl)
 				  << ": " << formatted_msg << Color::reset << std::endl;
+		_log_file << "[" << getTimestamp() << "] " << logLevelToStringFile(lvl)
+				  << ": " << formatted_msg << std::endl;
 	}
 
   private:
 	LogLevel _current_level;
 	std::string logLevelToString(LogLevel lvl);
+	std::string logLevelToStringFile(LogLevel lvl);
```
[the remark this answers is not in the document]
Comment (saladuit) [reply include/Logger.hpp:54]: const ref not possible cuz it makes a local var but the return type is const and the loglevel is const

**Pull request merged**